syscall/Makefile

```diff
@@ -76,27 +88,43 @@ $(BIN1): $(PROXY_OBJS)
 $(BIN2): $(STUB_OBJS)
 	$(call ARCHIVE, $@, $(STUB_OBJS))
 
+$(BIN3): $(WRAP_OBJS)
+	$(call ARCHIVE, $@, $(WRAP_OBJS))
+
+$(SYSCALLIST): .context
+
 .depend: Makefile $(SRCS)
-	$(Q) $(MKDEP) $(PROXYDEPPATH) $(STUBDEPPATH) \
+	$(Q) $(MKDEP) $(PROXYDEPPATH) $(STUBDEPPATH) $(WRAPDEPPATH) \
 	  "$(CC)" -- $(CFLAGS) -- $(SRCS) >Make.dep
 	$(Q) touch $@
 
 depend: .depend
 
 .context: syscall.csv
 	$(Q) $(MAKE) -C $(TOPDIR)$(DELIM)tools -f Makefile.host mksyscall
+ifeq ($(CONFIG_LIB_SYSCALL),y)
```

Comment:
The hook and syscall in flat build is mutual exclusive in the current implementation, but the syscall is also useful even in the flat build. How about we always use wrap mechanism for hook in all three build mode? The benefit is that:
1.The desgin and implementation is more simple
2.The hook and syscall can free combination

Reply:
Is it mean that the system call is processed as:
application -> syscall proxy -> (syscall handler) -> syscall stub -> **syscall wrapper** -> kernel function
in PROTECTED/KERNEL build?

If so, I'll try to make changes for it.

Reply:
Yes, then we use one solution to resolve all build mode.